### ISSUE TITLE
If I call pause multiple times, stack it and resume the persistor when the stack is empty.

### ISSUE THREAD

**Issue by skyshader** [rt2zz/redux-persist#432]
If I pause the persistor on every async call I do and resume on its complete / failure. If some of these async calls happens simultaneously, it just screws up the whole pause-resume sync.

What I want to achieve is that when I pause the persistor in one async call, the call to resume in that async call should resume the persistor if no other async call started simultaneously. And if they started, resume the persistor when the last async call finished.

Is there any other relevant solution for this case?

**Comment by rt2zz** [comment]
why do you want to pause the persistor during async calls? 

If you want to do this, I think you will have to maintain some registry of "async" calls, and only unpause when the registry is empty.

In redux-persist@5, there is currently no pause method. The architecture supports it, but I would want to better understand the use case first.

**Comment by skyshader** [comment]
Sorry for responding late.

My use case is that I I want to sync the storage across tabs for which I am using [redux-persist-crosstab](https://github.com/rt2zz/redux-persist-crosstab).

Now the async calls I have, also result in async state dispatches to redux which results in multiple updates in the localstorage.

I have 2 scenarios:
1. Multiple dispatches at the same time. Say setting authenticated to true, and also setting a global notification on the app for welcoming the user.
2. Dispatches occurring in 2 diff async calls, one after other.

The first dispatch and localstorage update triggers storage event to other tabs. When `redux-persist-crosstab` updates the localstorage on that specific tab, again storage event triggers. This creates an echo across tabs which does not create a problem if there are gaps between storage updates, otherwise creates inconsistency in the storage data if there are dispatches occurring simultaneously.

Hence I wanted to pause the persistor until I have updated all the necessary updates and then resume it to sync with storage only once.

I hope I explained my case clearly.

Thanks.
